From 12c140c2dbc1d64a18b00a9bbdfef91c50a53ef9 Mon Sep 17 00:00:00 2001
From: Philipp Kerling <pkerling@casix.org>
Date: Tue, 15 Jan 2019 12:46:52 +0100
Subject: [PATCH 1/2] [database] Log warning for MySQL < 5.6, MariaDB < 5.5

MySQL 5.5 is EOL, MariaDB 5.5 still supported but older versions are
not recommended
---
 xbmc/dbwrappers/mysqldataset.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/xbmc/dbwrappers/mysqldataset.cpp b/xbmc/dbwrappers/mysqldataset.cpp
index 115f53ffe7b6..7dff3e1b372e 100644
--- a/xbmc/dbwrappers/mysqldataset.cpp
+++ b/xbmc/dbwrappers/mysqldataset.cpp
@@ -176,8 +176,22 @@ int MysqlDatabase::connect(bool create_new) {
       static bool showed_ver_info = false;
       if (!showed_ver_info)
       {
-        CLog::Log(LOGINFO, "MYSQL: Connected to version %s", mysql_get_server_info(conn));
+        std::string version_string = mysql_get_server_info(conn);
+        CLog::Log(LOGINFO, "MYSQL: Connected to version {}", version_string);
         showed_ver_info = true;
+        unsigned long version = mysql_get_server_version(conn);
+        // Minimum for MySQL: 5.6 (5.5 is EOL)
+        unsigned long min_version = 50600;
+        if (version_string.find("MariaDB") != std::string::npos)
+        {
+          // Minimum for MariaDB: 5.5 (still supported)
+          min_version = 50500;
+        }
+
+        if (version < min_version)
+        {
+          CLog::Log(LOGWARNING, "MYSQL: Your database server version {} is very old and might not be supported in future Kodi versions. Please consider upgrading to MySQL 5.7 or MariaDB 10.2.", version_string);
+        }
       }
 
       // disable mysql autocommit since we handle it

From a0b269e8266f379703ea657aa85ee3ff1cbdfe4b Mon Sep 17 00:00:00 2001
From: Philipp Kerling <pkerling@casix.org>
Date: Tue, 15 Jan 2019 14:51:52 +0100
Subject: [PATCH 2/2] [database] Promote MySQL server version message to
 LOGNOTICE

---
 xbmc/dbwrappers/mysqldataset.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/dbwrappers/mysqldataset.cpp b/xbmc/dbwrappers/mysqldataset.cpp
index 7dff3e1b372e..be465afa2f02 100644
--- a/xbmc/dbwrappers/mysqldataset.cpp
+++ b/xbmc/dbwrappers/mysqldataset.cpp
@@ -177,7 +177,7 @@ int MysqlDatabase::connect(bool create_new) {
       if (!showed_ver_info)
       {
         std::string version_string = mysql_get_server_info(conn);
-        CLog::Log(LOGINFO, "MYSQL: Connected to version {}", version_string);
+        CLog::Log(LOGNOTICE, "MYSQL: Connected to version {}", version_string);
         showed_ver_info = true;
         unsigned long version = mysql_get_server_version(conn);
         // Minimum for MySQL: 5.6 (5.5 is EOL)
